### PR TITLE
Added a tab specifically for listing Silo content

### PR DESCRIPTION
**Pull request by jdfx**
Under corporation > starbases there is now a "Silos" tab that lists all the contents of all silo's owned by the corporation in one big list, opposed to clicking through each module.

This was a request from Black Omega Security (The OSS).

In order for it to work you also need to update the services repo to a modified version (another pull request to come).

Here is a modified composer.json if you want to test using my repos.

```
{
    "name": "eveseat/seat",
    "description": "Simple Eve Api Tool",
    "keywords": ["eve", "online", "corporation", "management", "api"],
    "license": "GPL-2.0",
    "type": "project",
    "require": {
        "php": ">=5.5.14",
        "laravel/framework": "5.1.*",
        "predis/predis": "~1.0",
        "eveseat/api": "1.*",
        "eveseat/console": "1.*",
        "eveseat/eveapi": "1.*",
        "eveseat/notifications": "1.*",
        "laravel/socialite": "~2.0"
    },
    "repositories": {
      "jdfx-eveseat-services": {
          "type": "package",
          "package": {
              "name": "eveseat/services",
              "description": "Adjusted services package for EVE seat.",
              "version": "1.0",
              "source": {
                  "url": "https://github.com/jdfx/services",
                  "type": "git",
                  "reference": "dee11fa62e69cc18f1ebb207d7b417ee0155e218"
              }
          }
        },
        "jdfx-eveseat-web": {
            "type": "package",
            "package": {
                "name": "eveseat/web",
                "description": "Adjusted web package for EVE seat.",
                "version": "1.0",
                "source": {
                    "url": "https://github.com/jdfx/web",
                    "type": "git",
                    "reference": "84c7eaa1822e458719c8df4d001e3bd18dc97df1"
                }
            }
        }
    },
    "require-dev": {
        "fzaninotto/faker": "~1.4",
        "mockery/mockery": "0.9.*",
        "phpunit/phpunit": "~4.0",
        "phpspec/phpspec": "~2.1"
    },
    "autoload": {
      "psr-4": {
          "App\\": "app/"
      },
      "files" : [
        "vendor/eveseat/services/src/Helpers/helpers.php"
      ],
        "classmap": [
            "database",
            "vendor/eveseat/services/src",
            "vendor/eveseat/web/src"
        ]
    },
    "autoload-dev": {
        "classmap": [
            "tests/TestCase.php"
        ]
    },
    "minimum-stability": "stable",
    "scripts": {
        "post-install-cmd": [
            "php artisan clear-compiled",
            "php artisan optimize"
        ],
        "post-update-cmd": [
            "php artisan clear-compiled",
            "php artisan optimize"
        ],
        "post-root-package-install": [
            "php -r \"copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "php artisan key:generate"
        ]
    },
    "config": {
        "preferred-install": "dist"
    }
}
```

### CODE REVIEW

**Comment by leonjza** [comment]
Thanks for the pr!

One can checkout the PR's from the repositories so there wont be a need for the `composer.json`, but thanks for the effort nonetheless 👍 

Ill get to testing this soon!

File: src/Http/Controllers/Corporation/ViewController.php [hunk 1]
```diff
@@ -188,6 +188,70 @@ public function getStarbases($corporation_id)
     }
 
     /**
+<<<<<<< HEAD
```
Comment (leonjza) [inline src/Http/Controllers/Corporation/ViewController.php]: Looks like an artifact from a merge conflict?

File: src/resources/views/corporation/starbase/ajax/silos-tab.blade.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                        {!! img('type', $starbase->starbaseTypeID, 64, ['class' => 'img-circle eve-icon small-icon']) !!}
+                        {{ $starbase->starbaseName }}
+                      </span>
+                    </td>
+
+                    @if($module['used_volume'] == 0 || $module['available_volume'] == 0)
+                    <?php $usedvolume = "0%"; ?>
+                    @else
+                    <?php  $usedvolume = number(100 * ($module['used_volume']) / ($module['available_volume']), 0)."%" ?>
+                    @endif
```
Comment (leonjza) [inline src/resources/views/corporation/starbase/ajax/silos-tab.blade.php]: This block needs to be calculated in the Controller. There should be no `<?php` tags in templates.

Comment (jdfx) [inline src/resources/views/corporation/starbase/ajax/silos-tab.blade.php]: No worries i'll clean this up.

File: src/Http/Controllers/Corporation/ViewController.php [hunk 3]
```diff
@@ -188,6 +188,70 @@ public function getStarbases($corporation_id)
...
+     * @param $corporation_id
+     *
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function getStarbaseSilos($corporation_id)
+    {
+
+        // not pretty but required
+        ini_set('max_execution_time', 12000);
```
Comment (leonjza) [inline src/Http/Controllers/Corporation/ViewController.php]: I have been wondering a bit about this. Should we not rather see if we can optimize the query somehow to not need a random `ini_set` here?

Comment (jdfx) [inline src/Http/Controllers/Corporation/ViewController.php]: Hmm, yes we can certainly try. It's likely never going to be *that speedy though. We should at the very least re-set the max_execution_time after this which looking at my code I haven't.

There's a couple of things you've spotted that I should of, thank you.

I'll get to updating these this week and see if we can make this sleek enough for merging.

Thanks Leonjza.

**Comment by leonjza** [comment]
Ping @jdfx

**Comment by jdfx** [comment]
Not forgotten, won't have chance to get to these for a couple of weeks but they will get done.

**Comment by leonjza** [comment]
Closing due to lack of feedback and braches being diverged quite a lot now.

**Comment by jdfx** [comment]
Apologies, I am in-undated with work currently. I will do this again on a newer branch as soon as I have time.

**Pull request closed without merging**